Check/Users/RolesList.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @file
+ * Contains \SiteAudit\Check\Users\RolesList.
+ */
+
+/**
+ * Class SiteAuditCheckUsersRolesList.
+ */
+class SiteAuditCheckUsersRolesList extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('List Roles');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Show all available roles and user counts.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {
+    $counts = array();
+    foreach ($this->registry['roles'] as $name => $count_users) {
+      $counts[] = "$name: $count_users";
+    }
+    return implode(', ', $counts);
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $roles = array_keys(\Drupal::entityManager()->getListBuilder('user_role')->load());
+    foreach ($roles as $role) {
```

Comment:
This is incredibly inefficient. Please use a `GROUP BY`. https://github.com/fluxsauce/site_audit/blob/7.x-1.x/Check/Users/RolesList.php#L56